google-cloud-logging/src/main/java/com/google/cloud/logging/LoggingImpl.java

```diff
@@ -478,10 +506,80 @@ private static WriteLogEntriesRequest writeLogEntriesRequest(LoggingOptions serv
   }
 
   public void write(Iterable<LogEntry> logEntries, WriteOption... options) {
-    get(writeAsync(logEntries, options));
+    if (inWriteCall.get() != null) {
+      return;
+    }
+    inWriteCall.set(true);
+
+    try {
+      writeLogEntries(logEntries, options);
+      for (LogEntry logEntry : logEntries) {
+        // flush pending writes if log severity at or above flush severity
+        if (logEntry.getSeverity().compareTo(flushSeverity) >= 0) {
+          flush();
+          break;
+        }
+      }
+    } finally {
+      inWriteCall.remove();
+    }
+  }
+
+  public void flush() {
+    // BUG(1795): We should force batcher to issue RPC call for buffered messages,
+    // so the code below doesn't wait uselessly.
+    ArrayList<ApiFuture<Void>> writesToFlush = new ArrayList<>();
+    synchronized (writeLock) {
+      writesToFlush.addAll(pendingWrites);
+    }
+
+   try {
+      ApiFutures.allAsList(writesToFlush).get();
+    } catch (InterruptedException|ExecutionException e) {
+       throw new RuntimeException(e);
```

Comment:
that would require marking the logging interface to throw the named exceptions, and I did n't want to introduce that. @garrettjonesgoogle/@pongad thoughts ?